codegen-serde/src/main/kotlin/software/amazon/smithy/rust/codegen/serde/Traits.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.serde
+
+import software.amazon.smithy.model.SourceLocation
+import software.amazon.smithy.model.node.Node
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.traits.AbstractTrait
+import software.amazon.smithy.model.traits.Trait
+import software.amazon.smithy.rust.codegen.core.util.orNull
+
+class SerdeTrait private constructor(
+    val serialize: Boolean,
+    val deserialize: Boolean,
+    val tag: String?,
+    val content: String?,
+    sourceLocation: SourceLocation,
+) :
+    AbstractTrait(ID, sourceLocation) {
+        override fun createNode(): Node {
+            val builder =
+                Node.objectNodeBuilder()
+                    .sourceLocation(sourceLocation)
+                    .withMember("serialize", serialize)
+                    .withMember("deserialize", deserialize)
+                    .withMember("tag", tag)
+                    .withMember("content", content)
+            return builder.build()
+        }
+
+        class Provider : AbstractTrait.Provider(ID) {
+            override fun createTrait(
+                target: ShapeId,
+                value: Node,
+            ): Trait {
+                val serialize = value.expectObjectNode().getBooleanMember("serialize").map { it.value }.orElse(true)
```

Comment:
Can we just do the check once? It would shorten the code a bit.
```suggestion
                value: ObjectNode,
            ): Trait {
                val serialize = value.getBooleanMember("serialize").map { it.value }.orElse(true)
```

Reply:
good idea, I refactored this entire file